modules/net-address/README.md

```diff
@@ -122,6 +122,27 @@ module "addresses" {
 # tftest modules=1 resources=1 inventory=psc.yaml e2e
 ```
 
+To create PSC address targeting a service provide `service_attachment`.
+```hcl
+module "addresses" {
+  source     = "./fabric/modules/net-address"
+  project_id = var.project_id
+  psc_addresses = {
+    cloudsql-one = {
+      address          = "10.0.16.32"
+      subnet_self_link = var.subnet.self_link
+      region           = var.region
+      service_attachment = {
+        name                        = "cloudsql-one"
```

Comment:
should we just reuse the address name for the service attachment name, and make `service_attachment` a string?

Reply:
Making `service_attachment` a string is problematic, because we have a condition `if psc.service_attachment != null` when creating forwarding rules. 

This results in error:
```
The "for_each" map includes keys derived from resource attributes that cannot be determined until apply, and so Terraform cannot determine the full set of keys that will identify the instances of this resource.
```

Because when trying to use this, service_attachment is known only after apply. So I'd keep it as an object, even if we ditch separate name for it.
```
     + cloudsql-one = {
          + address            = "10.0.16.32"
          + description        = "Terraform managed."
          + name               = null
          + network            = null
          + region             = "europe-west4"
          + service_attachment = (known after apply)
          + subnet_self_link   = "https://www.googleapis.com/compute/v1/projects/***/regions/europe-west4/subnetworks/e2e-test-1"

```

Reply:
makes total sense, we do the same in several other modules by adding "useless" keys to work around the same issue, sorry for giving you extra work :)